File: sdks/ios/Sources/AdxSDK/UserIdManager.swift
import Foundation

internal class UserIdManager {
    
    private let userDefaults = UserDefaults.standard
    private let userIdKey = "com.taskirx.sdk.userId"
    private let customUserIdKey = "com.taskirx.sdk.customUserId"
    
    // MARK: - User ID Management
    
    func getUserId() -> String {
        // Check for custom user ID first
        if let customId = userDefaults.string(forKey: customUserIdKey), !customId.isEmpty {
            return customId
        }
        
        // Get or generate SDK user ID
        if let userId = userDefaults.string(forKey: userIdKey), !userId.isEmpty {
            return userId
        }
        
        // Generate new user ID
        let newUserId = UUID().uuidString
        userDefaults.set(newUserId, forKey: userIdKey)
        return newUserId
    }
    
    func setCustomUserId(_ userId: String) {
        userDefaults.set(userId, forKey: customUserIdKey)
    }
    
    func clearCustomUserId() {
        userDefaults.removeObject(forKey: customUserIdKey)
    }
    
    func resetUserId() {
        userDefaults.removeObject(forKey: userIdKey)
        userDefaults.removeObject(forKey: customUserIdKey)
    }
}

File: sdks/android/src/main/kotlin/com/adxplatform/sdk/internal/UserIdManager.kt
package com.taskirx.sdk.internal

import android.content.Context
import android.content.SharedPreferences
import java.util.UUID

/**
 * Manages persistent user ID
 */
internal class UserIdManager(context: Context) {
    
    private val prefs: SharedPreferences = context.getSharedPreferences(
        "adx_sdk_prefs",
        Context.MODE_PRIVATE
    )
    
    companion object {
        private const val KEY_USER_ID = "user_id"
        private const val KEY_CUSTOM_USER_ID = "custom_user_id"
    }
    
    /**
     * Get or generate persistent user ID
     */
    fun getUserId(): String {
        // Check for custom user ID first
        val customId = prefs.getString(KEY_CUSTOM_USER_ID, null)
        if (!customId.isNullOrBlank()) {
            return customId
        }
        
        // Get or generate SDK user ID
        var userId = prefs.getString(KEY_USER_ID, null)
        if (userId.isNullOrBlank()) {
            userId = UUID.randomUUID().toString()
            prefs.edit().putString(KEY_USER_ID, userId).apply()
        }
        return userId
    }
    
    /**
     * Set custom user ID (from app)
     */
    fun setCustomUserId(userId: String) {
        prefs.edit().putString(KEY_CUSTOM_USER_ID, userId).apply()
    }
    
    /**
     * Clear custom user ID
     */
    fun clearCustomUserId() {
        prefs.edit().remove(KEY_CUSTOM_USER_ID).apply()
    }
}
